pkg/utils/pci_allocator.go

```diff
@@ -0,0 +1,83 @@
+package utils
+
+import (
+	"fmt"
+	"github.com/containernetworking/plugins/pkg/ns"
+	"os"
+	"path/filepath"
+)
+
+type PCIAllocation interface {
+	SaveAllocatedPCI(string, string) error
+	CleanAllocatedPCI(string) error
+	IsAllocated(string) error
+}
+
+type PCIAllocator struct {
```

Comment:
can you add Unit test for it ?

should be straight forward.

you migh just need to extend this to have a function that returns NetNS interface (`func GetNS(nspath string) (ns.NetNS, error)`)

which you will pass a mock in tests

Reply:
done I use the same method as in sriov_test package. 

it required root (to create the namespaces) but that was already in the project so my tests doesn't add a new requirement